sdcm/data_operations/data_operation.py

```diff
@@ -0,0 +1,194 @@
+                session.execute(query, timeout=300)
+
+    def query_one_partition(self, where_condition_to_delete, session):
+        query = f"select count(*) from %s where {where_condition_to_delete}"
+        self.log.debug("Run query: %s", query % self.schema_manage.table_name)
+        result = session.execute(query % self.schema_manage.table_name, timeout=240)
+        if result[0].count > 0:
+            raise ResurrectionDataFound(f"Found resurrection data in the table {self.schema_manage.table_name}. "
+                                        f"Partition: %s", where_condition_to_delete)
+
+        for view in self.schema_manage.view_names:
+            self.log.debug("Run query: %s", query % view)
+            result = session.execute(query % view, timeout=240)
+            if result[0].count > 0:
+                raise ResurrectionDataFound(f"Found resurrection data in the view {view}. Partition: %s",
+                                            where_condition_to_delete)
+
+    def validate_deleted_rows_not_resurrected(self, where_condition_to_delete):
+        with self.schema_manage.cluster.cql_connection_patient(self.schema_manage.cluster.nodes[0],
+                                                               keyspace=self.schema_manage.keyspace_name) as session:
+            for row in where_condition_to_delete:
+                self.query_one_partition(where_condition_to_delete=row, session=session)
+
+    def count_rows(self, entity):
+        try:
+            with self.schema_manage.tester.db_cluster.cql_connection_patient(self.schema_manage.tester.db_cluster.nodes[0],
+                                                                             keyspace=self.schema_manage.keyspace_name) as session:
+                query = f"select count(*) cnt from {entity}"
+                self.log.debug("Run select query: %s", query)
+                result = session.execute(query, timeout=300)
+                self.log.debug(
+                    "Rows in the %s.%s: %s", self.schema_manage.keyspace_name, self.schema_manage.table_name, result.one().cnt)
+                return result.one().cnt
+        except Exception as err:  # pylint: disable=broad-except
+            self.log.error("Failed query: %s. Error: %s", query, err)
+            return None
+
+
+# pylint: disable=too-many-instance-attributes
+class DataOperationThread:
+    # TODO: add description
+    """
+    Run thread with data manipulations during entire test. Sleep between operations.
+    For now one operation is supported: delete full partition
+    """
+    DEFAULT_INTERVAL_BETWEEN_OPERATIONS_SEC = 600
+
+    def __init__(self, tester, thread_name: str):
+        self.tester = tester
+        self.schema_manage = SchemaManage(tester=self.tester)
+        self.data_operations = DataOperations(schema_manage=self.schema_manage)
+        # TODO: maybe changed, expanded with new parameters
+        # parameter "data_operation_ks_class" defined as:
```

Comment:
why we need to pass down a specific part of the create keyspace command ? why not just us `pre_create_keyspace` and set the whole configuration we need ? and maybe only have configuration for which keyspace/table to use ?

Reply:
If keyspace should be defined by `pre_create_keyspace`, everybody that add `data_operation_ks_class` parameter has to remember to add `create keyspace` command to test yaml as well. Now he will need just define RF for the keyspace. 
Maybe in the future we will want to add more definitions